gpt_index/utils.py

```diff
@@ -114,24 +125,39 @@ def retry_on_exceptions_with_backoff(
 
     Args:
         lambda_fn (Callable): Function to be called and output we want.
-        exception_classes (List[Type[Exception]]): List of exception classes to retry.
+        exception_classes
+            (List[Union[Type[Exception], Tuple[Type[Exception], Callable]]]):
+            List of exception classes to retry, or a tuple with exception class
+            and a callable that takes that exception instance as input and returns
+            whether to retry. At least one needs to be provided.
         max_tries (int): Maximum number of tries, including the first. Defaults to 10.
         min_backoff_secs (float): Minimum amount of backoff time between attempts.
             Defaults to 0.5.
         max_backoff_secs (float): Maximum amount of backoff time between attempts.
             Defaults to 60.
 
     """
-    exception_class_tuples = tuple(exception_classes)
+    assert exception_classes, "At least one exception class needs to be provided"
```

Comment:
can we do raise ValueError (or some other error) instead? would prefer to keep asserts in unit tests